main.py

```diff
@@ -50,6 +50,23 @@ async def list_directory(directory: str = "") -> List[str]:
     return remote.list_directory(os.path.join(configs["master_path"], directory))
 
 
+@app.get("/args/")
+async def get_arguments(file: str) -> Dict[str, Dict[str, Any]]:
+    """Get arguments of the given experiment file and returns it.
```

Comment:
The function name and docstring say that this returns the arguments of the given experiment, but it actually returns more than that.
Is the extra information necessary? (For example, experiment class name, argument ui, scheduler defaults.)
Only two (name and arginfo) are described in the docstring.

I think you should choose one of the following:
1. Only return the arginfo.
2. Change the function name to, for example, `get_experiment_info()`.

Moreover, if you want to filter the `argument_ui` and `scheduler_defaults` out, you can use the response model here, I guess.
For example:
```python
import pydantic
class ExperimentInfo(pydantic.BaseModel):
  name: str
  arginfo: Dict[str, Any]

@app.get("/expinfo/", response_model=Dict[str, ExperimentInfo])
async def get_experiment_info(file: str) -> Any:
  remote = get_client(...)
  return remote.examine(file)
```

Reply:
Moreover, do you know why each argument is given in a list with the trailing `, null, null`?

Reply:
Oh, I was also thinking about the problem.

I think the experiment name is useful to show it in the builder GUI.
For your information, we can set the name as the first line of the docstring in the experiment class.
Please refer to the above code.

Then, I agree with you to change the function name to `get_experiment_info()` and filter the unnecessary data.
Thanks for suggesting the code😃

Reply:
> Moreover, do you know why each argument is given in a list with the trailing `, null, null`?

Honestly, I don't know what they mean.. I tried to find it, but lots of codes are dependent so I didn't😥

Reply:
Okay then! Please make the docstring clearer (describe the form in more detail of the contents).

Reply:
And also, how about changing the routing path to `/experiment/info`?
I want to add the prefix `/experiment/` to all functions about experiments such as `submit`.

Reply:
That's a great idea!